pkg/tsdb/cloudwatch/cloudwatch.go

```diff
@@ -65,20 +49,23 @@ func init() {
 type CloudWatchService struct {
 	LogsService *LogsService `inject:""`
 	Cfg         *setting.Cfg `inject:""`
+	sessions    *awsds.SessionCache
 }
 
 func (s *CloudWatchService) Init() error {
+	s.sessions = awsds.NewSessionCache()
 	return nil
 }
 
 func (s *CloudWatchService) NewExecutor(*models.DataSource) (plugins.DataPlugin, error) {
-	return newExecutor(s.LogsService, s.Cfg), nil
+	return newExecutor(s.LogsService, s.Cfg, s.sessions), nil
```

Comment:
A new executor is created for each new query, basically making the cache useless since it would be re-created every time. Instead I initialize the cache in CloudWatchService.Init. That way the cache will become global the way it has always been for the CloudWatch data source. We should consider making it non global, but that is something that should be done in a separate pr.